build_modules/lib/src/kernel_builder.dart

```diff
@@ -58,11 +58,17 @@ class KernelBuilder implements Builder {
   /// directory, which contains the platform kernel files.
   final String platformSdk;
 
+  /// The libraries file for the current platform.
+  ///
+  /// If not provided, defaults to "lib/libraries.json".
+  final String librariesPath;
+
   KernelBuilder(
       {@required this.platform,
       @required this.summaryOnly,
       @required this.sdkKernelPath,
       @required this.outputExtension,
+      this.librariesPath,
```

Comment:
nit: instead of allowing this to be null, assign it to the default in the initializer (same pattern as `platformSdk`)